File: frontend/src/api/ai.ts
import { axiosClient } from './axiosClient';

export interface GenerateVulnerabilityResponse {
  title: string;
  description: string;
  impact: string;
  recommendation: string;
  solution: string;
  severity: string;
  cvss_score: number;
  cvss_vector: string;
  cwe: string;
  owasp: string;
}

export const aiApi = {
  generateVulnerability: async (title: string): Promise<GenerateVulnerabilityResponse> => {
    const res = await axiosClient.post<GenerateVulnerabilityResponse>(
      '/ai/generate-vulnerability',
      { title }
    );
    return res.data;
  },
};

File: backend/app/schemas/ai.py
from decimal import Decimal
from pydantic import BaseModel, Field


class GenerateVulnerabilityRequest(BaseModel):
    title: str = Field(
        ...,
        min_length=3,
        max_length=255,
    )


class GenerateVulnerabilityResponse(BaseModel):
    title: str = Field(min_length=3, max_length=255)

    description: str = Field(min_length=10)

    impact: str = Field(min_length=10)

    recommendation: str = Field(min_length=10)

    solution: str = Field(min_length=10)

    severity: str = Field(min_length=3)

    cvss_score: Decimal = Field(ge=0, le=10)

    cvss_vector: str = Field(min_length=3)

    cwe: str = Field(min_length=3)

    owasp: str = Field(min_length=3)

class GenerateEvidenceResponse(BaseModel):
    evidence_description: str
